django/contrib/gis/geoip2/base.py

```diff
@@ -0,0 +1,239 @@
+import os
+import socket
+
+import geoip2.database
+
+from django.conf import settings
+from django.contrib.gis.geoip2.resources import City, Country
+from django.core.validators import ipv4_re
+from django.utils import six
+from django.utils.ipv6 import is_valid_ipv6_address
+
+# Creating the settings dictionary with any settings, if needed.
+GEOIP_SETTINGS = {
+    'GEOIP_PATH': settings.GEOIP_PATH if hasattr(settings, 'GEOIP_PATH') else None,
```

Comment:
More simply, I think we could use `getattr(settings, 'GEOIP_PATH', None)` and similar for the others.